### PR TITLE
Enabled dropout in the submodules; corrected DAG RNN architecture by …

### PR DESCRIPTION
**Pull request by erikhuck**
…adding activation function, batch norm, and dropout where needed

### CODE REVIEW

File: dna/models.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ class Submodule(nn.Module):
 
     def __init__(
         self, layer_sizes: typing.List[int], activation_name: str, use_batch_norm: bool, use_skip: bool = False, *,
-        device: str = 'cuda:0', seed: int = 0
+        device: str = 'cuda:0', seed: int = 0, dropout: float = 0.0
```
Comment (bjschoenfeld) [inline dna/models.py]: could you make dropout not a kw-only argument by putting it after `use_skip`

File: dna/models.py [hunk 2]
```diff
@@ -52,6 +52,7 @@ def __init__(
             for i in range(n_layers):
                 if i > 0:
                     layers.append(activation())
+                    layers.append(nn.Dropout(p=dropout))
```
Comment (bjschoenfeld) [inline dna/models.py]: only add a dropout layer if p > 0

File: dna/models.py [hunk 3]
```diff
@@ -404,6 +405,12 @@ def __init__(
         self.input_dropout = input_dropout
         self._input_submodule = self._get_input_submodule(output_size=hidden_state_size)
 
+        activation = ACTIVATIONS[activation_name]
+        self.activation = activation()
```
Comment (bjschoenfeld) [inline dna/models.py]: let's combine the above two lines into one

File: dna/models.py [hunk 4]
```diff
@@ -404,6 +405,12 @@ def __init__(
...
+        self.input_dropout_layer = nn.Dropout(p=input_dropout)
```
Comment (bjschoenfeld) [inline dna/models.py]: make sure the dropout layer is also moved to the correct device

File: dna/models.py [hunk 5]
```diff
@@ -447,6 +454,9 @@ def forward(self, args):
 
         # Pass the metafeatures through the input layer
         metafeatures = self._input_submodule(metafeatures)
+        metafeatures = self.activation(metafeatures)
+        metafeatures = self.input_dropout_layer(metafeatures)
+        metafeatures = self.batch_norm(metafeatures)
```
Comment (bjschoenfeld) [inline dna/models.py]: batch norm should only be used when `use_batch_norm == True`

File: dna/models.py [hunk 6]
```diff
@@ -447,6 +454,9 @@ def forward(self, args):
 
         # Pass the metafeatures through the input layer
         metafeatures = self._input_submodule(metafeatures)
+        metafeatures = self.activation(metafeatures)
+        metafeatures = self.input_dropout_layer(metafeatures)
```
Comment (bjschoenfeld) [inline dna/models.py]: dropout should only be used if `p > 0`
